emily/handler/tests/integration/deposit.rs

```diff
@@ -548,3 +548,129 @@ async fn update_deposits_updates_chainstate() {
+                bitcoin_txid: "test_fulfillment_bitcoin_txid".to_string(),
+                btc_fee: 2314,
+                stacks_txid: "test_fulfillment_stacks_txid".to_string(),
+            })));
+        }
+
+        apis::deposit_api::update_deposits(
+            &configuration,
+            UpdateDepositsRequestBody {
+                deposits: vec![DepositUpdate {
+                    bitcoin_tx_output_index: bitcoin_tx_output_index,
+                    bitcoin_txid: bitcoin_txid.into(),
+                    fulfillment,
+                    last_update_block_hash: "update_block_hash".into(),
+                    last_update_height: 34,
+                    status,
+                    status_message: "foo".into(),
+                }],
+            },
+        )
+        .await
+        .expect("Received an error after making a valid update deposit request api call.");
+
+        let response = apis::deposit_api::get_deposit(
+            &configuration,
+            bitcoin_txid,
+            &bitcoin_tx_output_index.to_string(),
+        )
+        .await
+        .expect("Received an error after making a valid get deposit api call.");
+        assert_eq!(response.bitcoin_txid, bitcoin_txid);
+        assert_eq!(response.status, status);
+
+        if should_reject {
+            apis::deposit_api::create_deposit(&configuration, create_deposit_body)
+                .await
+                .expect_err(&format!(
+                    "We should reject duplicate deposits, if old one is {:#?}",
+                    status
+                ));
+        } else {
+            apis::deposit_api::create_deposit(&configuration, create_deposit_body)
+                .await
+                .expect("Received an error after making a valid get deposit api call.");
+        }
+
+        let response = apis::deposit_api::get_deposit(
+            &configuration,
+            bitcoin_txid,
+            &bitcoin_tx_output_index.to_string(),
+        )
+        .await
+        .expect("Received an error after making a valid get deposit api call.");
+        assert_eq!(response.bitcoin_txid, bitcoin_txid);
+        let expected_status = if status == Status::Reprocessing {
+            Status::Pending
+        } else {
+            status
+        };
+        assert_eq!(response.status, expected_status);
+    }
+
+    // Maybe looks weird from the first glance, but I like that this won't compile if
+    // a new option to Status enum will be added without updating this test.
+    for status in [
+        Status::Accepted,
+        Status::Confirmed,
+        Status::Failed,
+        Status::Pending,
+        Status::Reprocessing,
+    ] {
+        match status {
+            Status::Accepted | Status::Confirmed | Status::Failed => {
+                test_duplicate(status, true).await;
+            }
+            Status::Pending | Status::Reprocessing => {
+                test_duplicate(status, false).await;
+            }
+        }
+    }
```

Comment:
We can use `test_case` for this to improve readability (both of code and test results), eg https://github.com/stacks-network/sbtc/blob/83b316c5d26a3434a1b53d558bc7f899ce6c03f2/emily/handler/src/database/entries/deposit.rs#L651-L657